### ISSUE TITLE
[Train] Training Loop with Weights & Biases

### ISSUE THREAD

**Issue by hongxuzhou** [hongxuzhou/illusion-of-state#5]
- Write a standard PyTorch training loop.
- Integrate wandb to log Loss and Accuracy.
- Save checkpoints.
